atlasdb-commons/src/test/java/com/palantir/util/crypto/Sha256HashTest.java

```diff
@@ -29,9 +34,54 @@ public void testSha256() {
         MessageDigest digest = Sha256Hash.getMessageDigest();
         assertThat(digest.getAlgorithm()).isEqualTo("SHA-256");
         assertThat(digest).isNotSameAs(Sha256Hash.getMessageDigest());
-        byte[] result = digest.digest("Hello World".getBytes(Charsets.UTF_8));
-        assertThat(BaseEncoding.base16().encode(result))
-                .isEqualTo("A591A6D40BF420404A011733CFB7B190D62C65BF0BCDA32B57B277D9AD9F146E");
-        assertThat(result).hasSize(32);
+        byte[] inputBytes = "Hello World".getBytes(Charsets.UTF_8);
+        byte[] sha256Bytes = digest.digest(inputBytes);
+        assertThat(sha256Bytes).hasSize(32);
+        assertThat(Sha256Hash.LOWER_CASE_HEX.encode(sha256Bytes))
+                .hasSize(64)
+                .isEqualToIgnoringCase("A591A6D40BF420404A011733CFB7B190D62C65BF0BCDA32B57B277D9AD9F146E")
+                .isEqualTo(expectedSha256(inputBytes))
+                .isEqualTo(legacyHex(sha256Bytes));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"test", "Hello, World", "12345678901234567890"})
+    public void testComputeHash(String input) {
+        byte[] inputBytes = input.getBytes(StandardCharsets.UTF_8);
+        String expectedSha256 = expectedSha256(inputBytes);
+
+        Sha256Hash hash = Sha256Hash.computeHash(inputBytes);
+        String legacyHexEncoded = legacyHex(
+                Sha256Hash.LOWER_CASE_HEX.decode(hash.serializeToHexString().toLowerCase(Locale.ROOT)));
+        assertThat(hash)
+                .isEqualTo(Sha256Hash.computeHash(inputBytes))
+                .hasSameHashCodeAs(Sha256Hash.computeHash(inputBytes))
+                .isEqualByComparingTo(Sha256Hash.computeHash(inputBytes))
+                .asString()
```

Comment:
nit: Can the things above the asString and those below be two separate assertions? It's a long enough chain that I missed the asString call and was (briefly) wondered how the types lined up!

Reply:
Especially as you seem to do (partially) the same assertions with hash.toString anyway, so I'm curious what the difference between the two sets is that you're aiming to defend against.

Reply:
I will make these assertions clearer.

I have a feeling the assertj excavator will end up rewriting `assertThat(hash.toString())` as `assertThat(hash).asString()` anyway.

Reply:
split the assertions up and also expanded the inputs